src/ex_getln.c

```diff
@@ -1915,6 +1915,11 @@ getcmdline_int(
 	    }
 	}
 
+	// Trigger CmdlineLeavePre autocommand
+	if (ccline.cmdfirstc != NUL && (c == '\n' || c == '\r' || c == K_KENTER
```

Comment:
The `ccline.cmdfirstc != NUL` condition doesn't look correct. When using `input()`, `ccline.cmdfirstc` is `NUL`, but shouldn't `CmdlineLeave` still be triggered in that case?

Reply:
https://github.com/vim/vim/pull/17190

Reply:
Thanks.